### PR TITLE
perf: add csharp ai generator template

### PR DESCRIPTION
**Pull request by SLdragon**

### CODE REVIEW

File: templates/csharp/custom-copilot-rag-api/OpenAPIClient/APIClientExceptions.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
+﻿using RestSharp;
+
+namespace OpenAPIClient
```
Comment (xzf0587) [inline templates/csharp/custom-copilot-rag-api/OpenAPIClient/APIClientExceptions.cs:3]: Should the namespace update to `{{SafeProjectName}}.OpenAPIClient` like other folder pattern?

File: templates/csharp/custom-copilot-rag-api/.gitignore [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# TeamsFx files
```
Comment (xzf0587) [inline templates/csharp/custom-copilot-rag-api/.gitignore:1]: the template folder name should be custom-copilot-custom-api like the name in js/python.
The fxcore will use the folder name for scafflolding.

File: templates/csharp/custom-copilot-rag-api/OpenAPIClient/APIClientExceptions.cs (hunk 1 above)
Comment (SLdragon) [reply templates/csharp/custom-copilot-rag-api/OpenAPIClient/APIClientExceptions.cs:3]: I want to make these code a standalone lib, so I just use a fixed namespace for it

File: templates/csharp/custom-copilot-rag-api/.gitignore (hunk 2 above)
Comment (SLdragon) [reply templates/csharp/custom-copilot-rag-api/.gitignore:1]: OK, sure, will update

File: templates/csharp/custom-copilot-rag-api/{{ProjectName}}.csproj.tpl [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+  <ItemGroup>
+    <Content Include="Prompts\chat\skprompt.txt">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+      <CopyToPublishDirectory>PreserveNewest</CopyToPublishDirectory>
+    </Content>
+  </ItemGroup>
+
+  <ItemGroup>
+    <Content Include="Prompts\chat\actions.json">
```
Comment (xzf0587) [inline templates/csharp/custom-copilot-rag-api/{{ProjectName}}.csproj.tpl:41]: It seems json file already included in Content Group. Not necessary to add them here.

Comment (SLdragon) [reply templates/csharp/custom-copilot-rag-api/{{ProjectName}}.csproj.tpl:41]: Sure, will fix it

**Review by xzf0587 — APPROVED** [review]

**Review by KennethBWSong — APPROVED** [review]

**Pull request merged**